lib/streamlit/caching/cache_utils.py

```diff
@@ -38,11 +42,17 @@
 _LOGGER = get_logger(__name__)
 
 
+@define
+class MsgData:
+    delta_type: str
+    message: Message
+
+
 class Cache:
     """Function cache interface. Caches persist across script runs."""
 
     @abstractmethod
-    def read_value(self, value_key: str) -> Any:
+    def read_value(self, value_key: str) -> Tuple[Any, List[MsgData]]:
```

Comment:
CachedValue could be moved into this module and this method changed to return it